doc/Language/compilation.pod6

```diff
@@ -0,0 +1,295 @@
+=end code
+
+Step 0 in getting us back out of this mess is to define a long name.
+A full module name in C<Raku> consists of the short-name, auth, version and API
+
+At the same time, the thing you install is usually not a single module but a distribution which probably contains one or more modules.
+Distribution names work just the same way as module names.
+Indeed, distributions often will just be called after their main module.
+An important property of distributions is that they are immutable.
+C<V< Foo:auth<nine>:ver<0.3>:api<1> >> will always be the name for exactly the same code.
+
+=head2 $*REPO
+In C<Perl> and C<Python> you deal with include paths, pointing to file system directories.
+In C<Raku> we call such directories "repositories" and each of these repositories is governed by an object that does the
+C<CompUnit::Repository> role.
+Instead of an C<B<@INC>> array, there's the C<$*REPO> variable.
+It contains a single repository object.
+This object has a B<next-repo> attribute that may contain another repository.
+In other words: repositories are managed as a I<linked list>.
+The important difference to the traditional array is, that when going through the list, each object has a say in whether
+to pass along a request to the next-repo or not.
+C<Raku> sets up a standard set of repositores, i.e. the "perl", "vendor" and "site" repositories, just like you know them from C<Perl>.
+In addition, we set up a "home" repository for the current user.
+
+Repositories must implement the C<need> method.
+A C<use> or C<require> statement in C<Raku> code is basically translated to a call to C<B<$*REPO>>'s C<need> method.
+This method may in turn delegate the request to the next-repo.
+
+=begin code
+role CompUnit::Repository {
+    has CompUnit::Repository $.next is rw;
+
+    method need(CompUnit::DependencySpecification $spec,
+                CompUnit::PrecompilationRepository $precomp,
+                CompUnit::Store :@precomp-stores
+                --> CompUnit:D
+                )
+        { ... }
+    method loaded(
+                --> Iterable
+                )
+        { ... }
+
+    method id( --> Str )
+        { ... }
+}
+=end code
+
+=head2 Repositories
+
+Rakudo comes with several classes that can be used for repositories.
+The most important ones are C<CompUnit::Repository::FileSystem> and C<CompUnit::Repository::Installation>.
+The FileSystem repo is meant to be used during module development and actually works just like C<Perl> when
+looking for a module.
+It doesn't support versions or auths and simply maps the short-name to a file system path.
+
+The Installation repository is where the real smarts are. When requesting a module, you will usually either do it
+via its exact long name, or you say something along the lines of "give me a module that matches this filter".
+Such a filter is given by way of a C<CompUnit::DependencySpecification> object which has fields for
+=item short-name,
```

Comment:
Undocumented, too. You can't link it, but please open an issue so that's it's documented or index it so that anyone looking for it arrives here.